examples/AdminClient/Program.cs

```diff
@@ -84,7 +95,218 @@ static async Task CreateTopicAsync(string bootstrapServers, string topicName)
                 catch (CreateTopicsException e)
                 {
-                    Console.WriteLine($"An error occured creating topic {e.Results[0].Topic}: {e.Results[0].Error.Reason}");
+                    Console.WriteLine($"An error occurred creating topic {e.Results[0].Topic}: {e.Results[0].Error.Reason}");
+                }
+            }
+        }
+
+        static List<AclBinding> ParseAclBindings(string[] args)
+        {
+            int nAclBindings = args.Length / 7;
+            var aclBindings = new List<AclBinding>();
+            for (int i = 0; i < nAclBindings; ++i)
+            {
+                var baseArg = i * 7;
+                var parseCorrect = true;
+                parseCorrect &= Enum.TryParse(typeof(ResourceType), args[baseArg], out object resourceType);
+                parseCorrect &= Enum.TryParse(typeof(ResourcePatternType), args[baseArg + 2], out object resourcePatternType);
+                parseCorrect &= Enum.TryParse(typeof(AclOperation), args[baseArg + 5], out object operation);
+                parseCorrect &= Enum.TryParse(typeof(AclPermissionType), args[baseArg + 6], out object permissionType);
+
+                if (!parseCorrect) return null;
+
+                var name = args[baseArg + 1];
+                var principal = args[baseArg + 3];
+                var host = args[baseArg + 4];
+
+                if (name == "") name = null;
+                if (principal == "") principal = null;
+                if (host == "") host = null;
+
+                aclBindings.Add(new AclBinding()
+                {
+                    Pattern = new ResourcePattern
+                    {
+                        Type = (ResourceType) resourceType,
+                        Name = name,
+                        ResourcePatternType = (ResourcePatternType) resourcePatternType
+                    },
+                    Entry = new AccessControlEntry
+                    {
+                        Principal = principal,
+                        Host = host,
+                        Operation = (AclOperation) operation,
+                        PermissionType = (AclPermissionType) permissionType
+                    }
+                });
+            }
+            return aclBindings;
+        }
+
+
+        static List<AclBindingFilter> ParseAclBindingFilters(string[] args)
+        {
+            var aclBindings = ParseAclBindings(args);
+            return aclBindings.Select(aclBinding => aclBinding.ToFilter()).ToList();
+        }
+
+
+        static void PrintAclBindings(List<AclBinding> aclBindings)
+        {
+            foreach (AclBinding aclBinding in aclBindings)
+            {
+                Console.WriteLine($"\t{aclBinding}");
+            }
+        }
+
+        static async Task CreateAclsAsync(string bootstrapServers, string[] commandArgs)
+        {
+            var numCommandArgs = commandArgs.Length;
+            var printUsage = numCommandArgs == 0 || numCommandArgs % 7 != 0;
```

Comment:
this logic better belongs in ParseAclBindings, and throw an exception if there's an issue. then handle that here to print usage.